File: SeaBattle/SeaBattle.CSharp/HumanPlayer.cs
using System.Drawing;

namespace SeatBattle.CSharp
{
    public class HumanPlayer : Player
    {
        private readonly Board _board;

        public HumanPlayer(string name, Board board)
            : base(name)
        {
            _board = board;
            _board.OnClick += OnBoardClick;

        }

        private void OnBoardClick(object sender, BoardCellClickEventErgs e)
        {
            if (PastShots.ContainsKey(new Point(e.X, e.Y)))
                return;

            ShotTargetChosen(e.X, e.Y);
        }

    }
}

File: SeaBattle/SeaBattle.Cpp/HumanPlayer.cpp
#include "stdafx.h"
#include "Player.h"
#include "HumanPlayer.h"
#include "Board.h"

HumanPlayer::HumanPlayer(String^ name, Board^ board) : Player(name)
{
	_board = board;
	_board->OnClick += gcnew EventHandler<BoardCellClickEventArgs^>(this,&HumanPlayer::OnBoardClick);
};

void HumanPlayer::OnBoardClick(Object^ sender, BoardCellClickEventArgs^ e)
{
	if (PastShots->ContainsKey(Point(e->X, e->Y)))
		return;

	ShotTargetChosen(e->X, e->Y);
};